### ISSUE TITLE
🐞 Bug glibc postinstall build-locale-archive does not work with kernel <= 5.4

### ISSUE THREAD

**Issue by 8leggedunicorn** [chromebrew/chromebrew#7872]
**Describe the bug**
Cannot successfully install the `glibc` package.

**To Reproduce**

1. I executed this command:

      ```shell
     issue `crew install glibc` or `crew build glibc`
      ```

      Output:

      ```shell
        chronos@localhost / $ crew install glibc
        glibc: The GNU C Library project provides the core libraries for GNU/Linux systems.
        https://www.gnu.org/software/libc/
        Version: 2.35
        License: LGPL-2.1+, BSD, HPND, ISC, inner-net, rc, and PCRE
        Performing pre-flight checks...
        Package glibc already installed, skipping...
        Performing post-install for glibc...
        Package glibc version is 2.35.
        System glibc version is 2.35.
        Creating symlinks to system glibc version to prevent breakage.
        Paring locales to a minimal set.
        /usr/local/lib/crew/lib/package.rb:305:in `system': Command failed with SIGABRT (signal 6) (core dumped): bash
        glibc failed to install: `CREW_DISABLE_ENV_OPTIONS="1" bash -c build-locale-archive` exited with 
        chronos@localhost / $
      ```
where running `build-locale-archive` by itself produces the following:

```shell
chronos@localhost / $ build-locale-archive
FATAL: kernel too old
Aborted (core dumped)
chronos@localhost / $
```

**Expected behavior**
Successfully install `glibc`

**Screenshots**

**Environment: Paste the output of `crew sysinfo -v` between the two `***` below**

***
<details><summary>Expand</summary>

- Architecture: `x86_64` (`x86_64`)
- Processor vendor: `GenuineIntel`
- User space: `64-bit`
- Chromebrew Kernel version: `5.4`
- Chromebrew Running in Container: `false`

- Chromebrew version: `1.30.9`
- Chromebrew prefix: `/usr/local`
- Chromebrew libdir: `/usr/local/lib64`

- Last update in local repository: d9057ac `Krb5 => 1.20.1 and update packages dependent upon krb5 (#7871) (17 hours ago)`

- OS variant: `Chrome OS`
- OS version: `dedede-release/R109-15236.66.0`
- OS channel: `stable-channel`

</details>
***

**Additional context**

```shell
chronos@localhost / $ uname -a
Linux localhost 5.4.223-19729-g00e2ddb12382 #1 SMP PREEMPT Wed Jan 11 19:56:30 PST 2023 x86_64 Intel(R) Celeron(R) N4500 @ 1.10GHz GenuineIntel GNU/Linux
```

This appears to be related to the additional conditional added to the glibc package 3 weeks ago: 

https://github.com/chromebrew/chromebrew/blob/master/packages/glibc.rb#L345-L365 with commit https://github.com/chromebrew/chromebrew/pull/7817

What I think needs to be done:

Let me know if I can help. My goal is to have a working glibc on my box, and through a package manager, ie Chromebrew, would be preferable. Thank you.

**Comment by satmandu** [comment]
Ok, so kernel 5.4 doesn't work for this either... fixing this shortly...

**Comment by satmandu** [comment]
Can you see if this works:
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=glibc_ver_fix CREW_TESTING=1 crew update
crew reinstall glibc
```

**Comment by 8leggedunicorn** [comment]
@satmandu first thank you that's amazingly fast work! Unfortunately that didn't seem to fix things:

using the updated version I'm still having the same issue:
***
<details><summary>Expand</summary>

- Architecture: `x86_64` (`x86_64`)
- Processor vendor: `GenuineIntel`
- User space: `64-bit`
- Chromebrew Kernel version: `5.4`
- Chromebrew Running in Container: `false`

- Chromebrew version: `1.30.9`
- Chromebrew prefix: `/usr/local`
- Chromebrew libdir: `/usr/local/lib64`

- Last update in local repository: c9466c2 `Change required kernel version to 5.10 for running postinstall build-locale-archive (#7874) (6 hours ago)`

- OS variant: `Chrome OS`
- OS version: `dedede-release/R109-15236.66.0`
- OS channel: `stable-channel`

</details>
***

this is after removing `glibc` and then attempting the install:

```
chronos@localhost ~ $ crew remove glibc
Glibc removed!
chronos@localhost ~ $ which build-locale-archive
which: no build-locale-archive in (/usr/local/bin:/usr/bin:/bin:/opt/bin:/usr/local/sbin)
chronos@localhost ~ $ crew install glibc
glibc: The GNU C Library project provides the core libraries for GNU/Linux systems.
https://www.gnu.org/software/libc/
Version: 2.35
License: LGPL-2.1+, BSD, HPND, ISC, inner-net, rc, and PCRE
Performing pre-flight checks...
Precompiled binary available, downloading...
Glibc archive downloaded.
Unpacking archive using 'tar', this may take a while...
Performing pre-install...
Performing install...
Using rdfind to convert duplicate files to hard links.
Now scanning ".", found 1433 files.
Now have 1433 files in total.
Removed 8 files due to nonunique device and inode.
Total size is 267104569 bytes or 255 MiB
Removed 979 files due to unique sizes from list. 446 files left.
Now eliminating candidates based on first bytes: removed 186 files from list. 260 files left.
Now eliminating candidates based on last bytes: removed 28 files from list. 232 files left.
Now eliminating candidates based on sha1 checksum: removed 232 files from list. 0 files left.
It seems like you have 0 files that are not unique
Totally, 0 B can be reduced.
Now making hard links.
Making 0 links.
Performing post-install for glibc...
Package glibc version is 2.35.
System glibc version is 2.35.
Creating symlinks to system glibc version to prevent breakage.
Paring locales to a minimal set.
/usr/local/lib/crew/lib/package.rb:305:in `system': Command failed with SIGABRT (signal 6) (core dumped): bash
glibc failed to install: `CREW_DISABLE_ENV_OPTIONS="1" bash -c build-locale-archive` exited with 
chronos@localhost ~ $ build-locale-archive 
FATAL: kernel too old
Aborted (core dumped)
chronos@localhost ~ $ 
```

let me know if there's a likely culprit and I'll go ahead and work on getting it to build, or let me know if there's something I can do to get it working. Thank you.

**Comment by 8leggedunicorn** [comment]
Also this isn't fixed at least for my build so let me know if you want me to create a new issue or if the status should be changed.

**Comment by satmandu** [comment]
Try `crew update ; crew reinstall glibc`

**Comment by 8leggedunicorn** [comment]
@satmandu hey I already did a `crew update`, sorry I should've added it to the cli output. So unfortunately the same outcome:

```
chronos@localhost ~ $ crew update ; crew reinstall glibc
From https://github.com/chromebrew/chromebrew
 * branch            master     -> FETCH_HEAD
HEAD is now at c9466c2 Change required kernel version to 5.10 for running postinstall build-locale-archive (#7874)
Package lists, crew, and library updated.
Checking for package updates...
Your software is up to date.
glibc: The GNU C Library project provides the core libraries for GNU/Linux systems.
Performing pre-flight checks...
Precompiled binary available, downloading...
Glibc archive downloaded.
Unpacking archive using 'tar', this may take a while...
Removing since upgrade or reinstall...
Glibc removed!
Performing pre-install...
Performing install...
Using rdfind to convert duplicate files to hard links.
Now scanning ".", found 1433 files.
Now have 1433 files in total.
Removed 8 files due to nonunique device and inode.
Total size is 267104569 bytes or 255 MiB
Removed 979 files due to unique sizes from list. 446 files left.
Now eliminating candidates based on first bytes: removed 186 files from list. 260 files left.
Now eliminating candidates based on last bytes: removed 28 files from list. 232 files left.
Now eliminating candidates based on sha1 checksum: removed 232 files from list. 0 files left.
It seems like you have 0 files that are not unique
Totally, 0 B can be reduced.
Now making hard links.
Making 0 links.
Performing post-install for glibc...
Package glibc version is 2.35.
System glibc version is 2.35.
Creating symlinks to system glibc version to prevent breakage.
Paring locales to a minimal set.
/usr/local/lib/crew/lib/package.rb:305:in `system': Command failed with SIGABRT (signal 6) (core dumped): bash
glibc failed to install: `CREW_DISABLE_ENV_OPTIONS="1" bash -c build-locale-archive` exited with 
chronos@localhost ~ $ 
```

**Comment by satmandu** [comment]
Just realized my math was wrong... Sigh... Hold on.

**Comment by satmandu** [comment]
For now just replace `return unless @libc_version.to_f >= 2.32 && CREW_KERNEL_VERSION.to_f >= 5.10` with `return` in `glibc.rb` and rerun `crew reinstall glibc`

**Comment by satmandu** [comment]
I'll test and submit a proper PR in the morning...

**Comment by satmandu** [comment]
Kernel version comparison in the GLIBC postinstall is broken ... 5.4 is greater than 5.10...

**Comment by 8leggedunicorn** [comment]
Ok, cool, appreciated that of course worked, and thank you!
